### PR TITLE
provider/aws: Retry RouteTable Route/Assocation creation

### PR DESCRIPTION
**Pull request by catsby**
Retry the creation of a Route Table Association. _Eventual Consistency Bandaid_ for #7104

### CODE REVIEW

**Comment by catsby** [comment]
```
TF_ACC=1 go test ./builtin/providers/aws -v -run=TestAccAWSRouteTable_ -timeout 120m
=== RUN   TestAccAWSRouteTable_importBasic
--- PASS: TestAccAWSRouteTable_importBasic (33.23s)
=== RUN   TestAccAWSRouteTable_basic
--- PASS: TestAccAWSRouteTable_basic (45.50s)
=== RUN   TestAccAWSRouteTable_instance
--- PASS: TestAccAWSRouteTable_instance (121.86s)
=== RUN   TestAccAWSRouteTable_tags
--- PASS: TestAccAWSRouteTable_tags (28.59s)
=== RUN   TestAccAWSRouteTable_vgwRoutePropagation
--- PASS: TestAccAWSRouteTable_vgwRoutePropagation (31.96s)
```

File: builtin/providers/aws/resource_aws_route_table.go [hunk 1]
```diff
@@ -301,7 +302,19 @@ func resourceAwsRouteTableUpdate(d *schema.ResourceData, meta interface{}) error
 			}
 
 			log.Printf("[INFO] Creating route for %s: %#v", d.Id(), opts)
-			if _, err := conn.CreateRoute(&opts); err != nil {
+			err := resource.Retry(1*time.Minute, func() *resource.RetryError {
+				_, err := conn.CreateRoute(&opts)
+				if err != nil {
+					if awsErr, ok := err.(awserr.Error); ok {
+						if awsErr.Code() == "InvalidRouteTableID.NotFound" && strings.Contains(awsErr.Message(), "does not exist") {
```
Comment (jbardin) [inline builtin/providers/aws/resource_aws_route_table.go]: Do we need the check for "does not exist" here? Seems like `InvalidRouteTableID.NotFound` should cover this case.

Comment (catsby) [inline builtin/providers/aws/resource_aws_route_table.go]: Yeah, probably don't need that. Removed in a6123ed

File: builtin/providers/aws/resource_aws_route_table_association.go [hunk 2]
```diff
@@ -40,11 +43,25 @@ func resourceAwsRouteTableAssociationCreate(d *schema.ResourceData, meta interfa
...
-	})
+	}
 
+	var resp *ec2.AssociateRouteTableOutput
+	var err error
+	err = resource.Retry(2*time.Minute, func() *resource.RetryError {
+		resp, err = conn.AssociateRouteTable(&associationOpts)
+		if err != nil {
+			if awsErr, ok := err.(awserr.Error); ok {
+				if awsErr.Code() == "InvalidRouteTableID.NotFound" && strings.Contains(awsErr.Message(), "does not exist") {
```
Comment (jbardin) [inline builtin/providers/aws/resource_aws_route_table_association.go]: sorry, ditto for this string check

**Comment by jbardin** [comment]
Other than the extra string comparisons, this LGTM provided travis passes.

File: builtin/providers/aws/resource_aws_route_table_association.go (hunk 2 above)
Comment (catsby) [inline builtin/providers/aws/resource_aws_route_table_association.go]: cleaned up in 75df603

**Pull request merged**